File: src/update_weather.py
# lambda function to update weather item in dynamodb
import json
import boto3
import os
dynamodb = boto3.resource('dynamodb')
table_name = os.environ.get("TABLE_NAME")
table = dynamodb.Table(table_name)

def lambda_handler(event, context):
    weather = json.loads(event['body'])['weather']
    town = json.loads(event['body'])['town']
    print(event)
    key = str(json.loads(event['pathParameters'])['id'])
    item = {
        'id': key,
        'weather': weather,
        'town': town
    }
    try:
        table.update_item(TableName=table_name, Item=item)
        return {
            'statusCode': 200,
            'message': 'Weather Updated Successfully!'
        }
    except table_name:
        return {
            'statusCode': 500,
            'message': 'Ooops! An error occured.'
        }

File: packages/functions/src/updateWeather.ts
import { DynamoDB } from "aws-sdk";
import { APIGatewayProxyEvent, APIGatewayProxyResult } from "aws-lambda";

const dynamoDb = new DynamoDB.DocumentClient();
const tableName = process.env.TABLENAME
export async function main(
  event: APIGatewayProxyEvent
): Promise<APIGatewayProxyResult> {
  const weatherId =
    event.pathParameters && event.pathParameters.id
      ? event.pathParameters.id
      : null;

  if (!weatherId) {
    return {
      statusCode: 404,
      body: JSON.stringify({ error: true }),
    };
  }

  if (event.body) {
    const data = {
      id: weatherId,
      weather: JSON.parse(event.body).weather?JSON.parse(event.body).weather: null,
      town: JSON.parse(event.body).town?JSON.parse(event.body).town: null,
    };

    const res = dynamoDb.put({TableName:tableName, Item: data}).promise()
    if(res == null)
      return {
        statusCode: 200,
        body: JSON.stringify({
          "message": "Weather updated successfully"
        }),
      };
    else
      return{
        statusCode: 404,
        body: JSON.stringify({
          message: "No weather data found",
        }),
      }
  }

  return {
    statusCode: 500,
    body: JSON.stringify({
      "message": "Weather failed to update"
    }),
  };
}